<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-08-20" />

<title>GranulomaMph</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">IntGranuloma</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GranulomaMph</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-08-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>IntGranuloma/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250717code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250717)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250717code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250717)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomLudewigLabIntGranulomatreea90907d33a3b88ec16743fa3058db3e77983b075targetblanka90907da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Ludewig-Lab/IntGranuloma/tree/a90907d33a3b88ec16743fa3058db3e77983b075" target="_blank">a90907d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomLudewigLabIntGranulomatreea90907d33a3b88ec16743fa3058db3e77983b075targetblanka90907da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Ludewig-Lab/IntGranuloma/tree/a90907d33a3b88ec16743fa3058db3e77983b075" target="_blank">a90907d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/GranulomaAll.Rmd
    Modified:   analysis/GranulomaMph.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GranulomaMph.Rmd</code>) and HTML
(<code>docs/GranulomaMph.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Ludewig-Lab/IntGranuloma/blob/ac89adb027080acdc727ef0636b7342b330460bb/analysis/GranulomaMph.Rmd" target="_blank">ac89adb</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-08-21
</td>
<td>
Aug25
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level4">
<h4>load packages</h4>
<pre class="r"><code>library(ExploreSCdataSeurat3)
library(runSeurat3)
library(Seurat)
library(ggpubr)
library(pheatmap)
library(SingleCellExperiment)
library(dplyr)
library(tidyverse)
library(viridis)
library(muscat)
library(circlize)
library(destiny)
library(scater)
library(metap)
library(multtest)
library(clusterProfiler)
library(org.Mm.eg.db)
library(msigdbr)
library(enrichplot)
library(DOSE)
library(grid)
library(gridExtra)
library(ggupset)
library(VennDiagram)
library(here)</code></pre>
</div>
<div id="load-merged-file-granuloma" class="section level4">
<h4>load merged file granuloma</h4>
<pre class="r"><code>basedir &lt;- here()
fileNam &lt;- paste0(basedir, &quot;/data/merged_allgranuloma_wofilter_seurat_wocl21.rds&quot;)
seurat &lt;- readRDS(fileNam)</code></pre>
</div>
<div id="set-color-vectors" class="section level4">
<h4>set color vectors</h4>
<pre class="r"><code>## col clustername
colclustername &lt;- c(&quot;#355C7D&quot;,&quot;#B1746FFF&quot;,&quot;#202547&quot;,&quot;#B09C85&quot;, &quot;#4e5a4c&quot;,&quot;#53354A&quot;,&quot;#2A363B&quot;,&quot;#8491B4FF&quot;,&quot;#00A087FF&quot;,&quot;#DC9989&quot;,&quot;#84ad83&quot;, &quot;#628395&quot;,&quot;#D33B44&quot;,&quot;#779d8d&quot;,&quot;#727077&quot;,&quot;#868686FF&quot;,&quot;#F8B195&quot;,&quot;#FF847C&quot;,&quot;#725663FF&quot;,&quot;#904D39&quot;,&quot;#91D1C2&quot;)
names(colclustername) &lt;- c(&quot;PdgfraloFb1&quot;,&quot;Mph1&quot;, &quot;CD8Tcell1&quot;,&quot;Neut&quot;, &quot;IgABcell1&quot;, &quot;LEC&quot;, &quot;Telocytes&quot;, &quot;DC&quot;, &quot;CD4Tcell/ILC2&quot;, &quot;Mph2&quot;,  &quot;NKcell&quot;, &quot;CD8Tcell2&quot;, &quot;Mph3&quot;, &quot;Trophocytes&quot;, &quot;PdgfrahiFb&quot;, &quot;IgABcell2&quot;, &quot;Myocytes&quot;, &quot;PdgfraloFb2&quot;, &quot;IgDBcell&quot;, &quot;Glial&quot;, &quot;ICC&quot;)

## col cond2
colcond2 &lt;- c(&quot;#202547&quot;,&quot;#BE3144&quot;)
names(colcond2) &lt;- c(&quot;WT&quot;, &quot;cko&quot;)</code></pre>
</div>
<div id="subset-mph" class="section level3">
<h3>subset Mph</h3>
<pre class="r"><code>seuratMph &lt;- subset(seurat, cluster_name %in% c(&quot;Mph1&quot;,&quot;Mph2&quot;,&quot;Mph3&quot;))
Idents(seuratMph) &lt;- seuratMph$cluster_name
levels(seuratMph)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph2&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>DimPlot(seuratMph, reduction = &quot;umap&quot;, cols = colclustername) </code></pre>
<p><img src="figure/GranulomaMph.Rmd/subset%20mph-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="abundances-mph" class="section level4">
<h4>abundances Mph</h4>
<pre class="r"><code>## make count list
datList &lt;- NULL
for(con in unique(seuratMph$cond2)){
  seuratSub &lt;- subset(seuratMph, cond2==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$cluster_name)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(cond2=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 29085  3707
[1] 29085  3073</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

ordXcond2 &lt;- c(&quot;WT&quot;, &quot;cko&quot;)

## plot abundance
ggbarplot(dat_all, x= &quot;cond2&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, legend = &quot;right&quot;, legend.titel = &quot;cluster_name&quot;, ylab = &quot;frequency&quot;, palette = colclustername, order = ordXcond2) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/relative%20abundance%20mph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mph1-vs-mph3" class="section level4">
<h4>Mph1 vs Mph3</h4>
<pre class="r"><code>seuratMph1and3 &lt;- subset(seurat, cluster_name %in% c(&quot;Mph1&quot;,&quot;Mph3&quot;))
Idents(seuratMph1and3) &lt;- seuratMph1and3$cluster_name
levels(seuratMph1and3)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph3&quot;</code></pre>
</div>
<div id="de-genes-mph-vs-mph3" class="section level4">
<h4>DE genes Mph vs Mph3</h4>
<pre class="r"><code>Idents(seuratMph1and3) &lt;- seuratMph1and3$cluster_name
levels(seuratMph1and3)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>DEGenesMph1and3 &lt;- FindAllMarkers(seuratMph1and3, only.pos=T, logfc.threshold = 0.2) %&gt;% filter(p_val_adj &lt; 0.01) </code></pre>
</div>
<div id="scatterplot-de-genes-mph1-vs-mph3" class="section level4">
<h4>scatterplot DE genes Mph1 vs Mph3</h4>
<pre class="r"><code>## table with expression levels of all genes
allID &lt;- seuratMph1and3
Idents(allID) &lt;- allID$cluster_name
avg.allID &lt;- AverageExpression(allID, group.by = &quot;cluster_name&quot;)
avg.allID &lt;- data.frame(log1p(avg.allID$RNA)) %&gt;% 
  rownames_to_column(var=&quot;gene&quot;)

## table with genes of a certain expression frequency
GeneFreq &lt;- as.data.frame(GetAssayData(allID, assay = &quot;RNA&quot;, slot = &quot;data&quot;)) %&gt;%
  tibble::rownames_to_column(var=&quot;gene&quot;) %&gt;% mutate(count=rowSums(.!=0)-1) %&gt;%
  mutate(countFreq=count/ncol(allID)) %&gt;%
  dplyr::select(gene, count, countFreq) %&gt;%
  dplyr::filter(countFreq&gt;=0.02)

## table with DEGenes to labeel - avg_log2FC &gt; 1.5 and p_val_adj &lt; 1e-100
genes.to.label &lt;- DEGenesMph1and3 %&gt;% group_by(.,cluster) %&gt;%
  dplyr::filter(p_val_adj &lt; 1e-100) %&gt;% dplyr::filter(avg_log2FC &gt; 1.5) %&gt;% mutate(labelNam=gsub(&quot;^.*\\.&quot;,&quot;&quot;,gene)) 
  
avg.allID &lt;- avg.allID %&gt;% mutate(labelNam=gsub(&quot;^.*\\.&quot;,&quot;&quot;,gene)) %&gt;% 
  mutate(colourGrp=ifelse(gene %in% genes.to.label$gene, &quot;lab&quot;, 
                          ifelse(gene %in% DEGenesMph1and3$gene, &quot;sig&quot;, &quot;notSig&quot;))) %&gt;% dplyr::filter(gene%in%GeneFreq$gene)

p_allID &lt;- ggscatter(avg.allID, x=&quot;Mph1&quot;, y=&quot;Mph3&quot;,
                     color = &quot;colourGrp&quot;,
                     palette = c(&quot;#B2182B&quot;, &quot;#808080&quot;, &quot;#2166AC&quot;),
                     label= &quot;labelNam&quot;,
                     label.select = genes.to.label$labelNam,
                     repel = T,
                     label.rectangle = F,
                     xlab = &quot;Mph1&quot;,
                     ylab = &quot;Mph3&quot;,
                     font.label = c(&quot;10&quot;, &quot;plain&quot;,&quot;black&quot;),
                     size=2,)
p_allID</code></pre>
<p><img src="figure/GranulomaMph.Rmd/scatter%20plot%20DEgenesMph1vsMph3-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-mph1" class="section level4">
<h4>GSEA Mph1</h4>
<pre class="r"><code>DEGenesMph1 &lt;- dplyr::filter(DEGenesMph1and3, cluster == &quot;Mph1&quot;)
DEGenesMph1 &lt;- DEGenesMph1 %&gt;% rownames_to_column(., var = &quot;long&quot;) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, long))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, long))

## GSEA Mph1
egoMph1 &lt;- enrichGO(gene = unique(DEGenesMph1$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoMph1 &lt;- setReadable(egoMph1, OrgDb = org.Mm.eg.db)
dotplot(egoMph1, showCategory=30)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/GSEA%20Mph1-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="gsea-mph3" class="section level4">
<h4>GSEA Mph3</h4>
<pre class="r"><code>DEGenesMph3 &lt;- dplyr::filter(DEGenesMph1and3, cluster == &quot;Mph3&quot;)
DEGenesMph3 &lt;- DEGenesMph3 %&gt;% rownames_to_column(., var = &quot;long&quot;) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, long))  %&gt;%
  mutate(EnsID=gsub(&quot;\\..*&quot;,&quot;&quot;, long))

## GSEA Mph3
egoMph3 &lt;- enrichGO(gene = unique(DEGenesMph3$EnsID),
                          OrgDb = org.Mm.eg.db,
                          keyType = &#39;ENSEMBL&#39;,
                          ont = &quot;BP&quot;,
                          pAdjustMethod = &quot;BH&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)
egoMph3 &lt;- setReadable(egoMph3, OrgDb = org.Mm.eg.db)
dotplot(egoMph3, showCategory=30)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/GSEA%20Mph3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-gsea" class="section level4">
<h4>barplot GSEA</h4>
<pre class="r"><code>## selected pathways
selMph1 &lt;- c(&quot;GO:0045088&quot;,&quot;GO:0019221&quot;, &quot;GO:0032640&quot;, &quot;GO:0034341&quot;)
egoMph1_fil &lt;- egoMph1%&gt;% filter(egoMph1@result$ID %in% selMph1)

egoMph1_fil_2 &lt;- pairwise_termsim(egoMph1_fil)
## make ggbarplot
p &lt;- ggbarplot(egoMph1_fil_2@result, x = &quot;Description&quot;, y = &quot;Count&quot;, fill = &quot;#B1746FFF&quot;, color=&quot;#B1746FFF&quot;, sort.val = &quot;asc&quot;, orientation = &quot;horizontal&quot;)
p</code></pre>
<p><img src="figure/GranulomaMph.Rmd/barplot%20GSEA-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cd45-positive-immmune-cells" class="section level4">
<h4>CD45 positive (immmune cells)</h4>
<pre class="r"><code>## subset CD45+ cells
seuratCD45pos &lt;- subset(seurat, cluster_name %in% c(&quot;CD8Tcell1&quot;,&quot;CD8Tcell2&quot;,&quot;NKcell&quot;,&quot;CD4Tcell/ILC2&quot;,&quot;Neut&quot;,&quot;Mph1&quot;,&quot;Mph2&quot;,&quot;Mph3&quot;,&quot;DC&quot;,&quot;IgABcell1&quot;,&quot;IgABcell2&quot;,&quot;IgDBcell&quot;))
table(seuratCD45pos$cluster_name)</code></pre>
<pre><code>
    CD8Tcell1     CD8Tcell2        NKcell CD4Tcell/ILC2          Neut          Mph1          Mph2 
         3007          1137          1502          1944          2477          4144          1766 
         Mph3            DC     IgABcell1     IgABcell2      IgDBcell 
          870          2048          2382           766           357 </code></pre>
</div>
<div id="average-heatmap-ccr1-ccr2-ccr5" class="section level4">
<h4>average heatmap Ccr1 Ccr2 Ccr5</h4>
<pre class="r"><code>Idents(seuratCD45pos) &lt;- seuratCD45pos$cluster_name
seurat &lt;- seuratCD45pos
levels(seurat)</code></pre>
<pre><code> [1] &quot;CD8Tcell1&quot;     &quot;CD8Tcell2&quot;     &quot;NKcell&quot;        &quot;CD4Tcell/ILC2&quot; &quot;Neut&quot;          &quot;Mph1&quot;         
 [7] &quot;Mph2&quot;          &quot;Mph3&quot;          &quot;DC&quot;            &quot;IgABcell1&quot;     &quot;IgABcell2&quot;     &quot;IgDBcell&quot;     </code></pre>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colclustername)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot &lt;- data.frame(gene=c(&quot;Ccr1&quot;, &quot;Ccr2&quot; ,&quot;Ccr5&quot;))

colVec &lt;- colclustername

avgHeatmap(seurat, selGenes = genesPlot, colVecIdent = colVec)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/avg%20heatmap%20Ccr1%20Ccr2%20Ccr5-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="average-heatmap-cytokine-mediated-signaling-pathway"
class="section level4">
<h4>average heatmap cytokine-mediated signaling pathway</h4>
<pre class="r"><code>Idents(seuratMph) &lt;- seuratMph$cluster_name
seurat &lt;- seuratMph
levels(seurat)</code></pre>
<pre><code>[1] &quot;Mph1&quot; &quot;Mph2&quot; &quot;Mph3&quot;</code></pre>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colclustername)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}

genesPlot &lt;- data.frame(gene=c(&quot;Osm&quot;, &quot;Il1b&quot;, &quot;Ccl6&quot;, &quot;Ccl9&quot;, &quot;Ccl2&quot;, &quot;Ccr1&quot;, &quot;Cxcl2&quot;, &quot;Ccl7&quot;, &quot;Ccl4&quot;, &quot;Il1rn&quot;,&quot;Myd88&quot;, &quot;Cxcl10&quot;, &quot;Cxcl3&quot;))

colVec &lt;- colclustername

avgHeatmap(seurat, selGenes = genesPlot, colVecIdent = colVec)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/avg%20heatmap%20cytokine-mediated%20signaling%20pathway-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="violin-plots-mph" class="section level3">
<h3>violin plots Mph</h3>
<div id="mph-subsets" class="section level4">
<h4>Mph subsets</h4>
<pre class="r"><code>Idents(seuratMph) &lt;- seuratMph$cluster_name
VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000058755.Osm&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000027398.Il1b&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000018927.Ccl6&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000019122.Ccl9&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000035385.Ccl2&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000025804.Ccr1&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000058427.Cxcl2&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000035373.Ccl7&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000018930.Ccl4&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000026981.Il1rn&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000032508.Myd88&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000034855.Cxcl10&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000029379.Cxcl3&quot;, pt.size = 0, cols = colclustername) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20Mph%20subset-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level4">
<h4>cond</h4>
<pre class="r"><code>Idents(seuratMph) &lt;- seuratMph$orig.ident
VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000058755.Osm&quot;, pt.size = 0, cols = colcond2, split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000027398.Il1b&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000018927.Ccl6&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000019122.Ccl9&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000035385.Ccl2&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000025804.Ccr1&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000058427.Cxcl2&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000035373.Ccl7&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000018930.Ccl4&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000026981.Il1rn&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000032508.Myd88&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000034855.Cxcl10&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(object=seuratMph, features = &quot;ENSMUSG00000029379.Cxcl3&quot;, pt.size = 0, cols = colcond2,split.by = &quot;cond2&quot;, split.plot = &quot;TRUE&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/GranulomaMph.Rmd/violin%20plot%20cond-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level4">
<h4>session info</h4>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri Aug 22 10:29:02 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1                  VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_25.1.1              org.Mm.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.1.0                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.5            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.5         
 [25] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.1.0           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-4           
 [64] reticulate_1.43.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.8         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] gridGraphics_0.5-1        fastDummies_1.7.5         lazyeval_0.2.2           
 [85] yaml_2.3.10               survival_3.8-3            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [91] progressr_0.15.1          tweenr_2.0.3              later_1.4.2              
 [94] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [97] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[100] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[103] TMB_1.9.17                spatstat.univar_3.1-4     mathjaxr_1.8-0           
[106] EnvStats_3.1.0            aplot_0.2.8               scatterplot3d_0.3-44     
[109] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[112] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[115] rbibutils_2.3             tools_4.4.0               globals_0.18.0           
[118] beeswarm_0.4.0            broom_1.0.9               nlme_3.1-168             
[121] lambda.r_1.2.4            assertthat_0.2.1          lme4_1.1-37              
[124] digest_0.6.37             numDeriv_2016.8-1.1       Matrix_1.7-3             
[127] farver_2.1.2              tzdb_0.5.0                remaCor_0.0.18           
[130] reshape2_1.4.4            yulab.utils_0.2.0         glue_1.8.0               
[133] cachem_1.1.0              polyclip_1.10-7           generics_0.1.4           
[136] Biostrings_2.72.1         mvtnorm_1.3-3             presto_1.0.0             
[139] parallelly_1.45.1         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-4             httr2_1.2.1              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.11.1              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.67.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-5      hms_1.1.3                
[172] fitdistrplus_1.2-4        cowplot_1.2.0             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.5.0             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.3            scatterpie_0.2.5         
[196] RSQLite_2.4.2             qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.1.0                edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5.1              MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.5-2   
[238] lifecycle_1.0.4           blme_1.0-6                S7_0.2.0                 
[241] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[244] robustbase_0.99-4-1       spatstat.geom_3.5-0       sn_2.1.1                 
[247] ggfun_0.2.0               future.apply_1.20.0       bslib_0.9.0              
[250] pillar_1.11.0             gplots_3.2.0              pcaMethods_1.96.0        
[253] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1                  VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_25.1.1              org.Mm.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.1.0                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.3.0               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.13             ggpubr_0.6.1                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.5            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.5         
 [25] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.1.0           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-4           
 [64] reticulate_1.43.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.8         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             ggrastr_1.0.2            
 [82] gridGraphics_0.5-1        fastDummies_1.7.5         lazyeval_0.2.2           
 [85] yaml_2.3.10               survival_3.8-3            scattermore_1.2          
 [88] crayon_1.5.3              RcppAnnoy_0.0.22          RColorBrewer_1.1-3       
 [91] progressr_0.15.1          tweenr_2.0.3              later_1.4.2              
 [94] ggridges_0.5.6            codetools_0.2-20          GlobalOptions_0.1.2      
 [97] aod_1.3.3                 KEGGREST_1.44.1           Rtsne_0.17               
[100] shape_1.4.6.1             limma_3.60.6              pkgconfig_2.0.3          
[103] TMB_1.9.17                spatstat.univar_3.1-4     mathjaxr_1.8-0           
[106] EnvStats_3.1.0            aplot_0.2.8               scatterplot3d_0.3-44     
[109] ape_5.8-1                 spatstat.sparse_3.1-0     xtable_1.8-4             
[112] car_3.1-3                 plyr_1.8.9                httr_1.4.7               
[115] rbibutils_2.3             tools_4.4.0               globals_0.18.0           
[118] beeswarm_0.4.0            broom_1.0.9               nlme_3.1-168             
[121] lambda.r_1.2.4            assertthat_0.2.1          lme4_1.1-37              
[124] digest_0.6.37             numDeriv_2016.8-1.1       Matrix_1.7-3             
[127] farver_2.1.2              tzdb_0.5.0                remaCor_0.0.18           
[130] reshape2_1.4.4            yulab.utils_0.2.0         glue_1.8.0               
[133] cachem_1.1.0              polyclip_1.10-7           generics_0.1.4           
[136] Biostrings_2.72.1         mvtnorm_1.3-3             presto_1.0.0             
[139] parallelly_1.45.1         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-4             httr2_1.2.1              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.11.1              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.67.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-5      hms_1.1.3                
[172] fitdistrplus_1.2-4        cowplot_1.2.0             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.5.0             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.3            scatterpie_0.2.5         
[196] RSQLite_2.4.2             qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.1.0                edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5.1              MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.5-2   
[238] lifecycle_1.0.4           blme_1.0-6                S7_0.2.0                 
[241] nloptr_2.2.1              sass_0.4.10               vctrs_0.6.5              
[244] robustbase_0.99-4-1       spatstat.geom_3.5-0       sn_2.1.1                 
[247] ggfun_0.2.0               future.apply_1.20.0       bslib_0.9.0              
[250] pillar_1.11.0             gplots_3.2.0              pcaMethods_1.96.0        
[253] locfit_1.5-9.12           jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
